src/Components/TodoContainer.js

```diff
@@ -1,74 +1,124 @@
-      }
-
-      // Parse the response JSON
-      const data = await response.json();
-
-      // Transform API data.records into todo objects
-      const todos = data.records.map((record) => ({
-        id: record.id,
-        title: record.fields.title,
-      }));
-
-      // Update todoList and isLoading states
-      setTodoList(todos);
-      setIsLoading(false);
-    } catch (error) {
-      console.log(`Fetch error: ${error.message}`);
+    const [todoList, setTodoList] = useState([]);
+    const [isLoading, setIsLoading] = useState(true);
+    const [isAscending, setIsAscending] = useState(true); // Track sorting order
+
+    // Memoize the fetchData function using useCallback to ensure a stable reference
+    const fetchData = useCallback(async () => {
+        // Define options for the API request
+        const options = {
+            method: "GET",
+            headers: {
+                Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`,
+            },
+        };
+
+        // Construct the URL for the API with ascending or descending order
+        // Update the URL with sort parameters (field and direction)
+        const order = isAscending ? "asc" : "desc";
+        const sortField = "created"; // Change this to the desired sorting field
+        const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}?sort[0][field]=${sortField}&sort[0][direction]=${order}`;
+
+        try {
+            const response = await fetch(url, options);
+
+            if (!response.ok) {
+                console.log("Airtable error response:");
+                console.log(response);
+                throw new Error(`Error: ${response.status}`);
+            }
+
+            // Parse the response JSON
+            const data = await response.json();
+            console.log('API Response Data:', data);
+
+            // Transform API data.records into todo objects
+            const todos = data.records.map((record) => ({
+                id: record.id,
+                title: record.fields.title,
+                createdTime: new Date(record.createdTime),
+            }));
+
+            // Update todoList and isLoading states
+            setTodoList(todos);
+            setIsLoading(false);
+        } catch (error) {
+            console.log(`Fetch error: ${error.message}`);
+        }
+    }, [isAscending]);
+
+    // Fetch data on initial render and whenever sorting order changes
+    useEffect(() => {
+        fetchData();
+        // eslint-disable-next-line 
+    }, []);
+
+    // Function to toggle sorting order and sort the todoList accordingly
+    function toggleSortingOrder() {
+        // Toggle between ascending and descending order
+        setIsAscending(!isAscending);
+
+        // Sort the todoList based on the new sorting order
+        setTodoList((prevTodoList) =>
+            [...prevTodoList].sort((a, b) => {
+                if (isAscending) {
```

Comment:
There's a small bug here where clicking the sort button sometimes doesn't work. I believe this is because we're changing two states at the same time, and setting state is technically an asynchronous operation that can take an unpredictable amount of time. 

Because we're doing two asynchronous things at the same time, I think this is sometimes causing a "Race condition" where we're not sure which one will finish first and we get different results depending on the outcome. 

Here I'm saving out the old value of the state before we try to change it, so both of the changes can be based on that stable, predefined value (rather than the unpredictable current state which may or may not have already been toggled by the time the sort code runs)

```suggestion
        const nextIsAscending = !isAscending;
        
        setIsAscending(nextIsAscending);

        // Sort the todoList based on the new sorting order
        setTodoList((prevTodoList) =>
            [...prevTodoList].sort((a, b) => {
                if (nextIsAscending) {
```